### PR TITLE
 fix Google Maps script irrelevant load

### PR DESCRIPTION
**Pull request by ddrmoscow**
Didn't open bug request, just code fix

Fixed issue # : 
New feature # :
Changed feature # :
Dev: 
Dev:

### CODE REVIEW

File: application/helpers/admin/statistics_helper.php [hunk 1]
```diff
@@ -3745,16 +3745,16 @@ public function generate_html_chartjs_statistics($surveyid, $allfields, $q2show
...
             $sSSL = 's';
         }
-        $sOutputHTML .= "<script type=\"text/javascript\" src=\"http{$sSSL}://maps.googleapis.com/maps/api/js?sensor=false$sGoogleMapsAPIKey\"></script>\n"
-        ."<script type=\"text/javascript\">var site_url='".Yii::app()->baseUrl."';var temppath='".Yii::app()->getConfig("tempurl")."';var imgpath='".Yii::app()->getConfig('adminimageurl')."';var aStatData=".ls_json_encode($aStatisticsData)."</script>";
+        $sGoogleMapsAPIKey = trim(Yii::app()->getConfig("googleMapsAPIKey"));
+        if ($sGoogleMapsAPIKey != '') {
+            $sGoogleMapsAPIKey = '&key='.$sGoogleMapsAPIKey;
+            $sOutputHTML .= "<script type=\"text/javascript\" src=\"http{$sSSL}://maps.googleapis.com/maps/api/js?sensor=false$sGoogleMapsAPIKey\"></script>\n";
+        }
+        $sOutputHTML .= "<script type=\"text/javascript\">var site_url='".Yii::app()->baseUrl."';var temppath='".Yii::app()->getConfig("tempurl")."';var imgpath='".Yii::app()->getConfig('adminimageurl')."';var aStatData=".ls_json_encode($aStatisticsData)."</script>";
```
Comment (Shnoulle) [inline application/helpers/admin/statistics_helper.php]: Did we still need this ?

File: application/helpers/userstatistics_helper.php [hunk 2]
```diff
@@ -2882,16 +2882,17 @@ public function generate_statistics($surveyid, $allfields, $q2show = 'all', $use
 
                 break;
             case 'html':
-                $sGoogleMapsAPIKey = trim(Yii::app()->getConfig("googleMapsAPIKey"));
-                if ($sGoogleMapsAPIKey != '') {
-                    $sGoogleMapsAPIKey = '&key='.$sGoogleMapsAPIKey;
-                }
                 $sSSL = '';
```
Comment (Shnoulle) [inline application/helpers/userstatistics_helper.php]: Can't we move this part to the test ?

File: application/helpers/userstatistics_helper.php [hunk 3]
```diff
@@ -2882,16 +2882,17 @@ public function generate_statistics($surveyid, $allfields, $q2show = 'all', $use
...
                 if (!empty($_SERVER['HTTPS']) && $_SERVER['HTTPS'] != "off") {
                     $sSSL = 's';
                 }
-                $sOutputHTML .= "<script type=\"text/javascript\" src=\"http{$sSSL}://maps.googleapis.com/maps/api/js?sensor=false$sGoogleMapsAPIKey\"></script>\n"
-                ."<script type=\"text/javascript\">var site_url='".Yii::app()->baseUrl."';var temppath='".Yii::app()->getConfig("tempurl")."';var imgpath='".Yii::app()->getConfig('adminimageurl')."';var aStatData=".ls_json_encode($aStatisticsData)."</script>";
+                $sGoogleMapsAPIKey = trim(Yii::app()->getConfig("googleMapsAPIKey"));
+                if ($sGoogleMapsAPIKey != '') {
+                    $sGoogleMapsAPIKey = '&key='.$sGoogleMapsAPIKey;
+                    $sOutputHTML .= "<script type=\"text/javascript\" src=\"http{$sSSL}://maps.googleapis.com/maps/api/js?sensor=false$sGoogleMapsAPIKey\"></script>\n";
```
Comment (Shnoulle) [inline application/helpers/userstatistics_helper.php]: `rc=\"//maps.googleapis.com/maps/api/js?sensor=false$sGoogleMapsAPIKey\">` must be OK, no need checking ssl or not (browser take current scheme).

File: application/helpers/userstatistics_helper.php (hunk 2 above)
Comment (Shnoulle) [reply application/helpers/userstatistics_helper.php]: Not checking ssl at all seem better.

**Comment by Shnoulle** [comment]
Question : with a valid google map API : it used only with Map question type ? Right ? It was loaded for all surey ?
Must check before if it work …

**Comment by cdorin93** [comment]
@Shnoulle do we have any bug report for this? I don't remember any gmaps-related issue :thinking:

**Comment by Shnoulle** [comment]
It's not an broke issue : script is loaded in statistics, even without MAP question type (or need google map).

And hhere : it just remove the map call if there are no google map API.

**Comment by cdorin93** [comment]
Ah, I see, thanks @Shnoulle 
@ddrmoscow , could you please rebase and solve the conflicts? Thanks for your contribution :)!

**Comment by c-schmitz** [comment]
Fixed in 19dab6760c18e948be4cf4f221b84b2cde298201

**Pull request closed without merging**